### DISCUSSION TITLE
How can I programmatically manage Kubernetes clusters with the Teleport API?

### DISCUSSION THREAD

**Discussion by RNHTTR** [gravitational/teleport#20374] [Q&A] [answered]
I'd like to programmatically manage Kubernetes clusters using the Teleport API. I'd prefer to do so via Python (i.e. without the [go client](https://goteleport.com/docs/api/introduction/#go-client). It [seems this must be possible](https://goteleport.com/docs/application-access/guides/api-access/)[^1] (demonstrated by accessing https://tele.{my-domain}/webapi/ping), but I don't see any reference as far as how to interact with the API, endpoints, etc.

[^1]: From the teleport docs linked above (emphasis mine): "Teleport Application Access can be used to access applications' (REST **or Teleport's own gRPC**) APIs with tools like [curl](https://man7.org/linux/man-pages/man1/curl.1.html) or Postman."

**Comment by webvictim** [comment, marked as answer]
The text on that documentation page seems to be a little misleading. Teleport does not expose a REST API. It does not have Python bindings for its GRPC API, or any way way to translate commands from Go to Python. 

The Teleport Go API docs and reference are located here:
- https://goteleport.com/docs/api/getting-started/
- https://pkg.go.dev/github.com/gravitational/teleport/api/client

**Reply by RNHTTR** [reply]
With the Go client, is it possible to interact directly with a Kubernetes pod? For example, can I delete a pod from the client?

**Reply by webvictim** [reply]
No - when you log into a Kubernetes cluster with Teleport, it basically gives you back certs and a Kubernetes context which will connect you to the cluster via Teleport. You'd then have to use a Kubernetes API client with that provided context to actually delete the pod.
